dao-modules/src/governance_voting.rs

```diff
@@ -0,0 +1,330 @@
+
+        call_contract(self.get_reputation_token_address(), "transfer_from", args)
+    }
+
+    fn burn_reputation(&mut self, owner: Address, amount: U256) {
+        let args: RuntimeArgs = runtime_args! {
+            "owner" => owner,
+            "amount" => amount,
+        };
+
+        call_contract(self.get_reputation_token_address(), "burn", args)
+    }
+
+    fn burn_creators_and_return_others_reputation(&mut self, voting_id: VotingId) {
+        let voters = self.voters.get(&voting_id);
+        for (i, address) in voters.iter().enumerate() {
+            let vote = self.votes.get(&(voting_id, address.unwrap_or_revert()));
+            if i == 0 {
+                // the creator
+                self.burn_reputation(self_address(), vote.stake);
+            } else {
+                // the voters - transfer from contract to them
+                self.transfer_reputation(self_address(), address.unwrap_or_revert(), vote.stake);
+            }
+        }
+    }
+
+    fn return_reputation(&mut self, voting_id: VotingId) {
+        let voters = self.voters.get(&voting_id);
+        for address in voters.iter() {
+            let vote = self.votes.get(&(voting_id, address.unwrap_or_revert()));
+            self.transfer_reputation(self_address(), address.unwrap_or_revert(), vote.stake);
+        }
+    }
+
+    fn redistribute_reputation(&mut self, voting: &Voting) {
+        let voters = self.voters.get(&voting.voting_id);
+        let total_stake = voting.stake_in_favor + voting.stake_against;
+        let mut transferred: U256 = U256::zero();
+        if voting.is_in_favor() {
+            for address in voters {
+                let vote = self
+                    .votes
+                    .get(&(voting.voting_id, address.unwrap_or_revert()));
+                if vote.choice {
+                    let to_transfer = total_stake * vote.stake / voting.stake_in_favor;
+                    self.transfer_reputation(
+                        self_address(),
+                        address.unwrap_or_revert(),
+                        to_transfer,
+                    );
+                    transferred += to_transfer;
+                }
+            }
+        } else {
+            for address in voters {
+                let vote = self
+                    .votes
+                    .get(&(voting.voting_id, address.unwrap_or_revert()));
+                if !vote.choice {
+                    let to_transfer = total_stake * vote.stake / voting.stake_against;
+                    self.transfer_reputation(
+                        self_address(),
+                        address.unwrap_or_revert(),
+                        to_transfer,
+                    );
+                    transferred += to_transfer;
+                }
+            }
+        }
+
+        // mark leftovers
+        if transferred < total_stake {
+            self.dust_amount
+                .set(self.get_dust_amount() + total_stake - transferred);
+        }
```

Comment:
```suggestion
        let dust = total_stake - transferred;
        if dust > 0 {
            self.dust_amount
                .set(self.get_dust_amount() + dust);
        }
```
That's more accurate, because it avoids overflow.